v03_pipeline/lib/reference_datasets/reference_dataset.py

```diff
@@ -0,0 +1,52 @@
+from enum import Enum
+
+from v03_pipeline.lib.model import AccessControl, DatasetType, Env, ReferenceGenome
+from v03_pipeline.lib.reference_data.config import CONFIG
+
+
+class ReferenceDataset(str, Enum):
+    cadd = 'cadd'
+    hgmd = 'hgmd'
+
+    @property
+    def _config(self) -> dict:
+        return CONFIG[self.value]
+
+    @classmethod
+    def for_reference_genome_dataset_type(
+        cls,
+        reference_genome: ReferenceGenome,
+        dataset_type: DatasetType,
+    ) -> list['ReferenceDataset']:
+        reference_datasets = {
+            (ReferenceGenome.GRCh37, DatasetType.SNV_INDEL): [
+                ReferenceDataset.cadd,
+                ReferenceDataset.hgmd,
+            ],
+            (ReferenceGenome.GRCh38, DatasetType.SNV_INDEL): [
+                ReferenceDataset.cadd,
+                ReferenceDataset.hgmd,
+            ],
+        }.get((reference_genome, dataset_type), [])
+        if not Env.ACCESS_PRIVATE_REFERENCE_DATASETS:
+            return [
+                rd
+                for rd in reference_datasets
+                if rd.access_control == AccessControl.PUBLIC
+            ]
+        return reference_datasets
+
+    @property
+    def access_control(self) -> AccessControl:
+        if self == ReferenceDataset.hgmd:
+            return AccessControl.PRIVATE
+        return AccessControl.PUBLIC
+
+    def version(self, reference_genome: ReferenceGenome) -> str:
+        return self._config[reference_genome]['version']
+
+    def raw_dataset_path(self, reference_genome: ReferenceGenome) -> str | list[str]:
+        return self._config[reference_genome]['raw_dataset_path']
+
+    def load_parsed_dataset_func(self) -> callable:
+        return self._config['load_parsed_dataset_func']
```

Comment:
what about just making these consistently named, so `cadd.py` defines `get_ht(...)` and we don't have a function in the config?

Reply:
we still need to make the function an attribute of the enum so that the task can call it, are you suggesting I go back to the dictionary access pattern?

Reply:
I was thinking about trying to do something similar to [what we did with the migrations](https://github.com/broadinstitute/seqr-loading-pipelines/blob/main/v03_pipeline/lib/migration/misc.py#L14), so that we could just return `cadd.get_ht` directly from this function.  But maybe that's too much!

Reply:
We could also try something like `importlib` to dynamically read in the correct file? Something like
```
module = imortlib.import_module(f'v03_pipeline.lib.reference_datasets.{self}')
return module. get_ht
```

Reply:
yeah, something like that is what I was thinking.  It's definitely doable and saves a config option.

Reply:
alright I tried the importlib approach, the only modification I made is that instead of returning a callable I just return the ht